### PR TITLE
feat: ⚓ adds geolocation API demo

### LINKED ISSUE #14: ✨ [New Demo Request]: Add the Geolocation Demo
Please include the request by answering these questions.

## What is the Example and Demo Topic?

Add a demo of the Geolocation API(https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API)

## Please explain why do you need it?

It is a most used web API. It would be nice to have an API Demo for this one

## Any Links, Reads?

https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API

## Do you want to contribute to the source code for this example?

Yes.

## If you are picking up this request to implement, PLEASE READ THIS

- [HOW TO ADD A DEMO](https://github.com/atapas/webapis-playground/blob/master/HOW-TO-ADD-DEMO.md)

### PR DESCRIPTION
**Pull request by wsameer**
# Description

This PR adds a demo of the Geolocation API. It is one of the most used web API. 
More information about the same can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API).

Fixes #14 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
There are no unit tests configured, so I have tested this manually.
One thing to notice, since I am adding a new demo, the "See Code" buttons redirects to 404 page. This will get auto fixed when the PR is merged and deployed.

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/atapas/webapis-playground/blob/master/HOW-TO-ADD-DEMO.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by wsameer** [comment]
@atapas I am done with the task.

**Comment by atapas** [comment]
> @atapas I am done with the task.

Thanks @wsameer, We will start the review soon and get back to you here.

**Comment by atapas** [comment]
@wsameer , Great work. I have seen the demo. Looks great. One question, Do you think we can embed the map in the demo page itself than opening it in a new tab? Is there a possibility?

File: web-apis/geolocation/index.js
Comment (atapas) [inline web-apis/geolocation/index.js:21]: @wsameer Can you please show the accuracy information as well?

**Comment by wsameer** [comment]
> @wsameer , Great work. I have seen the demo. Looks great. One question, Do you think we can embed the map in the demo page itself than opening it in a new tab? Is there a possibility?

@atapas Yeah, can be done. But it will add a lot of dependencies - [leaflet](https://react-leaflet.js.org/) . Also, there will be licensing issues, if that matters, since OpenStreetMap isn't MIT license, and Google Maps needs API key (you'll have to create one for the project being a maintainer).

If you're fine with all these, then I can work on the integration of that.

File: web-apis/geolocation/index.js
Comment (atapas) [inline web-apis/geolocation/index.js:31]: @wsameer , The `getCurrentPosition` method takes a few options as arguments. One of the important ones is `enableHighAccuracy: true`. Can we please pass that to get it more accurate?

Comment (atapas) [reply web-apis/geolocation/index.js:31]: To refer: https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition

**Review by atapas — CHANGES_REQUESTED** [review]
@wsameer A few minor reviews requested. Thanks!

**Comment by atapas** [comment]
> > @wsameer , Great work. I have seen the demo. Looks great. One question, Do you think we can embed the map in the demo page itself than opening it in a new tab? Is there a possibility?
> 
> @atapas Yeah, can be done. But it will add a lot of dependencies - [leaflet](https://react-leaflet.js.org/) . Also, there will be licensing issues, if that matters, since OpenStreetMap isn't MIT license, and Google Maps needs API key (you'll have to create one for the project being a maintainer).
> 
> If you're fine with all these, then I can work on the integration of that.

Sure. Thanks for the response.

**Comment by atapas** [comment]
> > > @wsameer , Great work. I have seen the demo. Looks great. One question, Do you think we can embed the map in the demo page itself than opening it in a new tab? Is there a possibility?
> > 
> > 
> > @atapas Yeah, can be done. But it will add a lot of dependencies - [leaflet](https://react-leaflet.js.org/) . Also, there will be licensing issues, if that matters, since OpenStreetMap isn't MIT license, and Google Maps needs API key (you'll have to create one for the project being a maintainer).
> > If you're fine with all these, then I can work on the integration of that.
> 
> Sure. Thanks for the response.

Let me understand the licensing implications. We can take it up as a separate issue.

File: web-apis/geolocation/index.js
Comment (wsameer) [reply web-apis/geolocation/index.js]: @atapas Added accuracy information in the demo

File: web-apis/geolocation/index.js
Comment (atapas) [inline web-apis/geolocation/index.js:41]: @wsameer , you might have to convert the number :)

image

File: utils/data/demos.js
Comment (atapas) [inline utils/data/demos.js:147]: Would be great to add a relevant emoji in the title like the one we have for others?

https://emojipedia.org/

**Review by atapas — CHANGES_REQUESTED** [review]
A couple of more to get done!

File: utils/data/demos.js
Comment (wsameer) [reply utils/data/demos.js:147]: Okay, it wasn't mentioned in the requirements. Do you have any suggestions for which emoji you want to add ?

File: web-apis/geolocation/index.js
Comment (wsameer) [reply web-apis/geolocation/index.js:41]: @atapas convert? you mean make it a whole number? or anything else?

File: utils/data/demos.js
Comment (atapas) [reply utils/data/demos.js:147]: U can use this one: https://emojipedia.org/anchor/

File: web-apis/geolocation/index.js
Comment (atapas) [reply web-apis/geolocation/index.js:41]: At least,
- 2 decimals after a point.
- It would be great to short the number like the above number is 2.46kms

This may help: https://www.npmjs.com/package/short-number

Comment (wsameer) [reply web-apis/geolocation/index.js:41]: @atapas  okay, so you want to convert into Kilometers. got it!

**Review by atapas — APPROVED** [review]
@wsameer You rock!!! All Good

**Comment by atapas** [comment]
Hi @wsameer 

Thanks a lot for taking the issue and making the app better. Please feel free to pick up more, contribute more. I wish you a great #Hactoberfest ahead. You will be added to the contribution list right after we merge this PR. Don't forget to share about your contribution publicly and provide your support with a star ⭐.

Great work 🔥 !!!!

**Comment by atapas** [comment]
@all-contributors please add @wsameer  for Code

**Pull request merged**
